tests/core/test_subdir_data.py

```diff
@@ -264,8 +266,53 @@ def test_use_only_tar_bz2(platform=OVERRIDE_PLATFORM):
         assert precs[0].fn.endswith(".conda")
 
 
+def test_subdir_data_coverage(platform=OVERRIDE_PLATFORM):
+    class ChannelCacheClear:
+        def __enter__(self):
+            return
+
+        def __exit__(self, *exc):
+            Channel._cache_.clear()
+
+    with ChannelCacheClear(), make_temp_env(), env_vars(
+        {"CONDA_PLATFORM": platform},
+        stack_callback=conda_tests_ctxt_mgmt_def_pol,
+    ):
+        channel = Channel(url_path(join(CHANNEL_DIR, platform)))
+
+        sd = SubdirData(channel)
+        sd.load()
+        assert all(isinstance(p, PackageRecord) for p in sd._package_records[1:])
+
+        assert all(r.name == "zlib" for r in sd._iter_records_by_name("zlib"))  # type: ignore
+
+        sd.reload()
+
+        assert all(r.name == "zlib" for r in sd._iter_records_by_name("zlib"))  # type: ignore
+
+        # newly deprecated, run them anyway
+        sd._save_state(sd._load_state())
```

Comment:
```suggestion
def test_subdir_data_coverage(request, platform=OVERRIDE_PLATFORM):
    request.addfinalizer(Channel._cache_.clear)

    with make_temp_env(), env_vars(
        {"CONDA_PLATFORM": platform},
        stack_callback=conda_tests_ctxt_mgmt_def_pol,
    ):
        channel = Channel(url_path(join(CHANNEL_DIR, platform)))

        sd = SubdirData(channel)
        sd.load()
        assert all(isinstance(p, PackageRecord) for p in sd._package_records[1:])

        assert all(r.name == "zlib" for r in sd._iter_records_by_name("zlib"))  # type: ignore

        sd.reload()

        assert all(r.name == "zlib" for r in sd._iter_records_by_name("zlib"))  # type: ignore

        # newly deprecated, run them anyway
        sd._save_state(sd._load_state())
```

Reply:
I think the standard Python feature is easier to understand without consulting the pytest documentation, and I'd rather not wait another hour for the tests to re-run.